mlflow/telemetry/events.py

```diff
@@ -320,3 +321,24 @@ def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
 
 class AutologgingEvent(Event):
     name: str = "autologging"
+
+
+class TraceSource(str, Enum):
+    """Source of a trace received by the MLflow server."""
+
+    MLFLOW_PYTHON_CLIENT = "MLFLOW_PYTHON_CLIENT"
+    UNKNOWN = "UNKNOWN"
+
+
+class TracesReceivedByServerEvent(Event):
+    name: str = "traces_received_by_server"
+
+    @classmethod
+    def parse(cls, arguments: dict[str, Any]) -> dict[str, Any] | None:
+        source = arguments.get("source")
+        if "count" not in arguments:
+            raise ValueError("count is required for TracesReceivedByServerEvent")
+        return {
+            "source": source.value if isinstance(source, TraceSource) else source,
+            "count": arguments["count"],
+        }
```

Comment:
```suggestion
```
We don't need this since TracesReceivedByServerEvent is not used in decorator, parse method is only used when the event is used as `@record_usage_event(event_name)` :)